enterprise/reporting/setup.go

```diff
@@ -12,43 +12,63 @@ import (
 )
 
 type Factory struct {
-	EnterpriseToken   string
-	Log               logger.Logger
-	once              sync.Once
-	reportingInstance types.ReportingI
+	EnterpriseToken     string
+	Log                 logger.Logger
+	once                sync.Once
+	reportingInstance   types.ReportingI
+	edReportingInstance types.ReportingI
 }
 
 // Setup initializes Suppress User feature
-func (m *Factory) Setup(backendConfig backendconfig.BackendConfig) types.ReportingI {
+func (m *Factory) Setup(backendConfig backendconfig.BackendConfig) types.ReportingInstances {
 	if m.Log == nil {
 		m.Log = logger.NewLogger().Child("enterprise").Child("reporting")
 	}
 	m.once.Do(func() {
 		reportingEnabled := config.GetBool("Reporting.enabled", types.DefaultReportingEnabled)
 		if !reportingEnabled {
 			m.reportingInstance = &NOOP{}
+			m.edReportingInstance = &NOOP{}
 			return
 		}
 
 		if m.EnterpriseToken == "" {
 			m.reportingInstance = &NOOP{}
+			m.edReportingInstance = &NOOP{}
 			return
 		}
 
 		h := NewFromEnvConfig(m.Log)
+		ed := NewEdReporterFromEnvConfig()
 		rruntime.Go(func() {
```

Comment:
rruntime.Go(func() {
			go h.setup(backendConfig)
                        go ed.setup(backendConfig)
		})
		
		not possible?

Reply:
I just didn't want to run them in same go-routines.

Reply:
rruntime.Go(func() {->  isn't this same?

Reply:
I think we are not going ahead with `go mymethod()` approach in recent changes in `rudder-server`
Hence I have used `rruntime.Go` multiple times to signify we are running our specific methods in two different go-routines.